robot_smach_states/src/robot_smach_states/util/designators/ed_designators.py

```diff
@@ -120,39 +116,41 @@ def __init__(self, robot, type="", center_point=None, radius=0, id="", parse=Tru
         @param id the ID of the object to get info about
         @param parse whether to parse the data string associated with the object model or entity
         @param criteriafuncs a list of functions that take an entity and return a bool (True if criterium met)
-        @param weight_function returns a weight for each entity, the one with the lowest weight will be selected (could be a distance calculation)
+        @param weight_function returns a weight for each entity, the one with the lowest weight will be selected
+        (could be a distance calculation)
         @param type_designator same as type but dynamically resolved trhough a designator. Mutually exclusive with type
-        @param center_point_designator same as center_point but dynamically resolved trhough a designator. Mutually exclusive with center_point
+        @param center_point_designator same as center_point but dynamically resolved trhough a designator.
+        Mutually exclusive with center_point
         @param id_designator same as id but dynamically resolved through a designator. Mutually exclusive with id"""
         super(EdEntityDesignator, self).__init__(resolve_type=Entity, name=name)
+
+        assert not type or type_designator is None, "Specify either type or type_designator, not both"
+        assert center_point is None or type_designator is None, \
```

Comment:
Should type_designator be center_point_designator here?

Reply:
Well spotted